spec/models/comment_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  before :all do
+    @author = User.create(name: 'Okari Nyandika', photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+                          bio: 'Student at Microverse.')
+    @user = User.create(name: 'Samson Opiyo', photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+                        bio: 'Teacher from Poland.')
+    @post = Post.create(author: @author, title: 'Okari ', text: 'Hello World!', comments_counter: 0,
+                        likes_counter: 0)
+  end
+
+  context '#create' do
+    it 'is valid with existing user and post' do
+      expect(Comment.new(user: @user, post: @post)).to be_valid
+    end
+
+    it 'is not valid without post' do
+      expect(Comment.new(user: @user)).to_not be_valid
+    end
+
+    it 'is not valid without user' do
+      expect(Comment.new(post: @post)).to_not be_valid
+    end
+  end
+
+  context '#update_comments_counter' do
```

Comment:
- [ ] Please make sure to test all your methods in your User model for example the `update_comments_counter` method.

Reply:
I don't have such a method in my user model

Reply:
Hello @2004-okari , I meant to say the Comment model. That was a typo on my part. Sorry about that. These are the methods seen in your Comment model:
![image](https://github.com/2004-okari/BLOG_APP/assets/45460192/7ec399ab-f223-467e-b51a-cb07ce0b6de9)

Reply:
Hello @Lucas-Masaba ,  I also have tests for the mentioned method
```
ontext '#update_comments_counter' do
    before :all do
      5.times { |comment_i| Comment.create(user: @user, post: @post, text: (comment_i + 1).to_s) }
    end

    it 'keeps track of comments and equals 5' do
      expect(@post.comments_counter).to eq 5
    end
  end
```